### PR TITLE
DOC: Reindexing behaviour of dataframe column-assignment missing

### PR DESCRIPTION
**Pull request by hewittk**
- [X] xref #39845 (.loc[] portion of issue)
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by hewittk** [comment]
Do we need a whatsnew entry for a documentation-only issue? If so, where and in what general format to we put it?

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -443,6 +443,35 @@ def loc(self) -> _LocIndexer:
...
+        Assign column using Series
+
+        >>> df.loc[:, 'shield'] = pd.Series({7: 8, 8: 10, 9: 13})
+        >>> df
+           max_speed  shield
+        7          1       8
+        8          4      10
+        9          7      13
+
+        Assign partial column with labels in dataframe missing from Series
```
Comment (phofl) [inline pandas/core/indexing.py:457]: Could you add something like the Series is aligned to the Index of the DataFrame

**Comment by phofl** [comment]
Don't think we need a whatsnew, you can not use closes if you are fixing part of the issue, because this would close the issue. E.g. your number is wrong, you are referencing the pr

File: pandas/core/indexing.py (hunk 1 above)
Comment (hewittk) [reply pandas/core/indexing.py]: Does what I added now in the latest commit look like what you were asking for? If not, can you clarify what to put a bit more?

Comment (phofl) [reply pandas/core/indexing.py]: I thought about something like Assigning a Series with non matching Index aligns the right hand side to the Index of the left.

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -443,6 +443,36 @@ def loc(self) -> _LocIndexer:
         8          4       5
         9          7       8
 
+        **Setting values using Series**
```
Comment (jreback) [inline pandas/core/indexing.py:446]: i guess its ok to do this, but this is not what user's should be doing, so add a big warning that you you should simply be using `df['shield'] = ....` as the preferred syntax.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by phofl — APPROVED** [review]

File: pandas/core/indexing.py (hunk 2 above)
Comment (hewittk) [reply pandas/core/indexing.py:446]: @jreback Does the warning I've added now look sufficient? If not, how do you recommend changing it?

Comment (jreback) [reply pandas/core/indexing.py:446]: actually this duplicates an example higher up. would prefer the warning in the Notes section (you can also add a comment before the example above)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by hewittk** [comment]
> This pull request is stale because it has been open for thirty days with no activity. Please update or respond to this comment if you're still interested in working on this.

I apologize for the lack of activity. I was very busy with finals and will be working towards closing this pull request in the coming days.

File: pandas/core/indexing.py [hunk 3]
```diff
@@ -287,6 +287,10 @@ def loc(self) -> _LocIndexer:
         - An alignable boolean Series. The index of the key will be aligned before
           masking.
         - An alignable Index. The Index of the returned selection will be the input.
+
+          .. warning:: Preferred syntax to assign a column: `df[]` instead of
```
Comment (mroeschke) [inline pandas/core/indexing.py:291]: Is this true? I didn't find note of this in the original issue?

File: pandas/core/indexing.py [hunk 4]
```diff
@@ -388,6 +392,9 @@ def loc(self) -> _LocIndexer:
 
         **Setting values**
 
+          .. warning:: Preferred syntax to assign a column: `df['shield']`
```
Comment (mroeschke) [inline pandas/core/indexing.py:395]: Same

**Comment by mroeschke** [comment]
Sorry this went un-reviewed for a while. Could you merge master again to this branch?

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. Let us know if you're still interested in working on this and we can reopen. Closing.

**Pull request closed without merging**